optaplanner-examples/src/main/java/org/optaplanner/examples/common/swingui/ConstraintMatchesDialog.java

```diff
@@ -56,6 +56,14 @@ public ConstraintMatchesDialog(SolverAndPersistenceFrame solverAndPersistenceFra
         this.solutionBusiness = solutionBusiness;
     }
 
+    @Override
+    public void setVisible(boolean visible) {
+        if (!visible) {
+            setContentPane(new JPanel()); // force redraw on visible(true)
```

Comment:
setVisible() shouldn't mess with the contentPane.
I suspect this works because it calls Container.validate() indirectly.

IIRC, there are 2 methods in Swing to fix these kinds of bugs:
 repaint(); // typically with a custom paint() method this needs to be called, otherwise it's overkill
 validate(); (or invalidate()?) // typically changing something invalidates the layout, and Swing wants to wait until all your changes are done before calculating the new layout. validate() is used to signal to Swing that you're done. IIRC...